source/index.ts

```diff
@@ -53,27 +53,25 @@ export default class PQueue<QueueType extends Queue<RunFunction, EnqueueOptionsT
 
 	private readonly _throwOnTimeout: boolean;
 
-	constructor(options?: Options<QueueType, EnqueueOptionsType>) {
+	constructor(passedOptions?: Options<QueueType, EnqueueOptionsType>) {
 		super();
 
-		// eslint-disable-next-line @typescript-eslint/no-object-literal-type-assertion
-		options = {
+		const options: Options<QueueType, EnqueueOptionsType> = {
 			carryoverConcurrencyCount: false,
 			intervalCap: Infinity,
 			interval: 0,
 			concurrency: Infinity,
 			autoStart: true,
 			queueClass: PriorityQueue,
-			...options
-		// TODO: Remove this `as`.
-		} as Options<QueueType, EnqueueOptionsType>;
+			...passedOptions
+		} as any;
```

Comment:
Is this really any different/better than what was already there?

Reply:
It was just to avoid disabling the lint rule. I'll revert this one too.